### PR TITLE
util: add env var AZURE_FP_SERVICE_PRINCIPAL_ID

### PR DESCRIPTION
**Pull request by troy0820**
### Which issue this PR addresses:

Fixes [10128055](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/10128055)

### What this PR does / why we need it:

FF-int fails when trying to list the service principals (permissions) to get the object ID.  This PR will query the object ID from the RP-Config and set the necessary parameters without having to use the sp app id to get the object id.  

### Test plan for issue:

Current unit test/e2e tests and deploying FF-int in a pipeline. 

### Is there any documentation that needs to be updated for this PR?
No

### CODE REVIEW

File: pkg/util/cluster/aad.go
Comment (jewzaam) [inline pkg/util/cluster/aad.go:30]: Is this specific to USGovCloud?

**Review by jewzaam — CHANGES_REQUESTED** [review]
Also need to handle using objectID for deleteRoleAssignments.

**Review by jewzaam — DISMISSED** [review]

**Comment by jewzaam** [comment]
Tests are timing out, https://github.com/Azure/ARO-RP/pull/1547 to increase timeout.

**Review by m1kola — APPROVED** [review]

**Review by m1kola — DISMISSED** [review]
Sorry, in previous review selected wrong button. I wanted to put this on hold before we discuss https://github.com/Azure/ARO-RP/pull/1546/files#r651764183

File: pkg/util/cluster/cluster.go
Comment (troy0820) [reply pkg/util/cluster/cluster.go:154]: This env var `AZURE_FP_SERVICE_PRINCIPAL_ID` is what we should be doing instead of having to check if it doesn't exist or not.  I think the idea of the fallback to the `getServicePrincipal` was to ensure some fail-safe way of providing the fpSPID if this isn't available (which shouldn't happen).  

As far as standardizing this approach across environments, I believe we should do that but the best method forward for that should be discussed further outside of this PR. @m1kola

Comment (jewzaam) [reply pkg/util/cluster/cluster.go:154]: We could take out the lookup.  We always pass in this value, it should never be empty.

**Comment by troy0820** [comment]
Manually ran pipeline with env var `AZURE_FP_SERVICE_PRINCIPAL_ID` set to "int" and results are [here](https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=43573321&view=logs&j=10f66467-59e5-53bc-0c8a-42cc89770de4&t=10f66467-59e5-53bc-0c8a-42cc89770de4).  

E2E passed but want to make sure that plumbing is vetted for `fpServicePrincipalId` in code changes.  @jewzaam

**Review by jewzaam — APPROVED** [review]

**Comment by jewzaam** [comment]
@m1kola I'm dismissing your review as the comment is addressed.  If that is incorrect please let me know and we can address in a new PR.  We have time before we cut a new RP release and my understanding is we're good on this guy now but.. lmk!

**Comment by m1kola** [comment]
All looks good to me. Thanks for the heads up!

**Pull request merged**
